### PR TITLE
Fix BasicSwap FakeRun Typo

### PR DESCRIPTION
**Pull request by evmckinney9**
### Summary
Resolve https://github.com/Qiskit/qiskit-terra/issues/10147

### Details and comments
Reference the function instead of the attribute.

### CODE REVIEW

**Comment by evmckinney9** [comment]
I checked other files, didn't see a similar mistake.

**Comment by Cryoris** [comment]
Thanks for the fix! Could you add a test that verifies this behavior is fixed now (in [the basic swap test file](https://github.com/Qiskit/qiskit-terra/blob/main/test/python/transpiler/test_basic_swap.py)) and a releasenote documenting this fix? See e.g. the [contributing guidelines](https://github.com/Qiskit/qiskit-terra/blob/main/CONTRIBUTING.md) for some more details and let us know if we can help you with that!

**Comment by 1ucian0** [comment]
Thanks for the fix! Do you mind adding a test to show the bug that this is fixing?

File: test/python/transpiler/test_basic_swap.py [hunk 1]
```diff
@@ -368,6 +368,51 @@ def test_far_swap_with_gate_the_middle(self):
...
+        expected.swap(qr[2], qr[1])
+        expected.cx(qr[1], qr[0])
+        expected.h(qr[1])
+        expected.cx(qr[0], qr[1])
+
+        pass_ = BasicSwap(coupling, fake_run=True)
+        after = pass_.run(dag)
+
+        self.assertNotEqual(circuit_to_dag(expected), after)
+        self.assertEqual(dag, after)
```
Comment (jakelishman) [inline test/python/transpiler/test_basic_swap.py:415]: This test mostly looks good to me, thanks.  One small addition, just to check that `fake_run` is functioning correctly: could we modify it slightly so we check the output `property_set` for the `"final_layout"` key, and check it comes out the same between the `fake_run=True` and `fake_run=False` cases?

It'd use the same circuit as your test, but with the assertions looking something along the lines of:
```python
circuit = ...

fake_pm = PassManager([BasicSwap(coupling, fake_run=True)])
real_pm = PassManager([BasicSwap(coupling, fake_run=False)])

self.assertEqual(circuit, fake_pm.run(circuit))
self.assertNotEqual(circuit, real_pm.run(circuit))
self.assertIsInstance(fake_pm.property_set["final_layout"], Layout)
self.assertEqual(fake_pm.property_set["final_layout"], real_pm.property_set["final_layout"])
```
(I may have made some mistakes in that.)

It tests the same "fake run doesn't crash" and "fake run doesn't modify the input" properties as yours, but also ensures that the logical 'output' of the fake run is emitted too.

**Review by jakelishman — COMMENTED** [review]
Thanks for looking at this!  I have one thing about the test, but after that we'll be good to merge.

**Comment by evmckinney9** [comment]
Oops, seems I messed things about moving around my fork's branches.

**Comment by jakelishman** [comment]
No worries, it's not a problem.  In case you weren't sure, this is why it's not generally recommended to make PRs from your `main` branch - GitHub will let you because of its fork+PR process, but it's pretty easy to get the branches tangled.

**Pull request closed without merging**